client/state/data-layer/wpcom/read/site/posts/follow/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @format
+ */
+
+/**
+ * External Dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import { READER_CONVERSATION_FOLLOW } from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { unfollowConversation } from 'state/reader/conversations/actions';
+import { bypassDataLayer } from 'state/data-layer/utils';
+
+export function requestConversationFollow( { dispatch }, action ) {
+	dispatch(
+		http( {
+			method: 'POST',
+			apiNamespace: 'wpcom/v2',
+			path: `/read/sites/${ action.payload.blogId }/posts/${ action.payload.postId }/follow`,
+			body: {}, // have to have an empty body to make wpcom-http happy
+			onSuccess: action,
+			onFailure: action,
+		} )
+	);
+}
+
+export function receiveConversationFollow( store, action, response ) {
+	// validate that it worked
+	const isFollowing = !! ( response && response.success );
+	if ( ! isFollowing ) {
```

Comment:
@dmsnell recently added something that handles redirecting to the error case very graceful

We can now validate the reponse against a json-schema:

```js
import { dispatchRequest, makeParser } from '...';

// success must be true pass
const conversationFollowSchema = {
  "properties": {
    "success": { enum: [ true ] }
  }
};

...

dispatchRequest( _, _, _, {
  fromApi: makeParser( conversationFollowSchema )
});
```

if the json schema doesn't validate then it throws and will redirect to the error handler

Reply:
That's really smart, but for someone who reads this code in the future, I think it's much less clear to see what's happening compared to:

```
const isFollowing = !! ( response && response.success );
if ( ! isFollowing ) {
	receiveConversationFollowError( store, action );
	return;
}
```

Reply:
sure, also note that we don't have to `makeParser` as well, which can be nice for simple "is this validly doing what I expect…"

```js
const fromApi = ( { success } ) => {
	if ( ! success ) {
		throw new Error( 'could not follow conversation' );
	}

	return true;
}
```

Not a whole lot different, but at least lets us keep our success and error handlers more semantically specific…

```js
dispatchRequest( notifySuccess, updateAndNotifiyFailure, { fromApi } )
```

cc: @samouri I think it's time we start allowing `onSuccess` and `onFailure` to be basic actions so we can do this:

```js
dispatchRequest( successNotice( 'Yay' ), successNotice( 'Boo' ), { fromApi } )
```